### PR TITLE
[dhcp_relay] optimize dhcp relay cli

### PR DESCRIPTION
**Pull request by Xichen96**
#### Why I did it
Dhcp implementation included inefficient code

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Using redis.keys with pattern instead of looping over all keys.

#### How to verify it
Run unit tests.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py [hunk 1]
```diff
@@ -146,6 +156,7 @@ def keys(self, pattern='*'):
 
 
 swsssdk.interface.DBInterface._subscribe_keyspace_notification = _subscribe_keyspace_notification
+swsssdk.interface.DBInterface.keys = keys
```
Comment (lguohan) [inline dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py:159]: are we still using swsssdk? @qiluo-msft , i thought swsssdk has been deprecated already.

**Comment by Blueve** [comment]
This PR contains test update only, I think you should have function code change accordingly. Are you forgot push them?

File: dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py (hunk 1 above)
Comment (Blueve) [reply dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py:159]: Seems like the `swsssdk` only be used in test. (macsec test use same approach as well: https://github.com/sonic-net/sonic-buildimage/blob/40834d51ee9b31ebf89b58b176b7fa7773d6d3f3/dockers/docker-macsec/cli-plugin-tests/mock_tables.py#L8)
We are using `swsscommon` in code implementation.

@yaqiangz for vis, can we use swsscommon in test?

**Comment by Xichen96** [comment]
> This PR contains test update only, I think you should have function code change accordingly. Are you forgot push them?

@Blueve Updated.

File: dockers/docker-dhcp-relay/cli/show/plugins/show_dhcp_relay.py [hunk 2]
```diff
@@ -49,9 +49,8 @@ def __init__(self):
     def get_interface(self):
         """ Get all names of all interfaces in DHCPv6_COUNTER_TABLE """
         vlans = []
-        for key in self.db.keys(self.db.STATE_DB):
-            if DHCPv6_COUNTER_TABLE in key:
-                vlans.append(key[21:])
+        for key in self.db.keys(self.db.STATE_DB, DHCPv6_COUNTER_TABLE + "|*"):
```
Comment (Blueve) [inline dockers/docker-dhcp-relay/cli/show/plugins/show_dhcp_relay.py:52]: How about use `self.table_name + '*'` ?

File: dockers/docker-dhcp-relay/cli/show/plugins/show_dhcp_relay.py [hunk 3]
```diff
@@ -49,9 +49,8 @@ def __init__(self):
...
+            vlans.append(key[21:])
```
Comment (Blueve) [inline dockers/docker-dhcp-relay/cli/show/plugins/show_dhcp_relay.py:53]: How to explain this magic number?
Can we use define it in __init__ method with calculation? such as `len(self.table_name)`?

File: dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py (hunk 1 above)
Comment (yaqiangz) [reply dockers/docker-dhcp-relay/cli-plugin-tests/mock_tables.py:159]: swsssdk is used in mock dbconnector in CLI ut (sonic-utilities / dhcp_relay / dhcp_server / macsec). From code, seems the purpose of them is to mock dbconnector in swsscommon with swsssdk
```
from swsssdk import SonicDBConfig, SonicV2Connector, ConfigDBConnector, ConfigDBPipeConnector
from swsscommon import swsscommon
...
swsscommon.SonicV2Connector = SonicV2Connector
swsscommon.ConfigDBConnector = ConfigDBConnector
swsscommon.ConfigDBPipeConnector = ConfigDBPipeConnector
```
And these changes were introduced when replacing swsssdk with swsscommon. swsssdk in running code has been removed, but in ut it is preserved for mocking. @qiluo-msft could you please help to check whether it's expected and whether we should remove swsssdk in CLI ut?
https://github.com/sonic-net/sonic-utilities/pull/1217/files#diff-cbaa80f38565f63f1f982f2220cc6e90d5a0b21028604886d87bd77f093e1ba5
https://github.com/sonic-net/sonic-utilities/pull/1392/files#diff-cbaa80f38565f63f1f982f2220cc6e90d5a0b21028604886d87bd77f093e1ba5

**Review by Blueve — APPROVED** [review]

**Comment by Xichen96** [comment]
@yxieca Hi Ying, pls help review

**Pull request merged**